internal/kibana/synthetics/schema.go

```diff
@@ -1,15 +1,299 @@
+				MarkdownDescription: "Whether the monitor is enabled. Default: `true`",
+			},
+			"tags": schema.ListAttribute{
+				ElementType:         types.StringType,
+				Optional:            true,
+				MarkdownDescription: "An array of tags.",
+			},
+			"alert": monitorAlertConfigSchema(),
+			"service_name": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "The APM service name.",
+			},
+			"timeout": schema.Int64Attribute{
+				Optional:            true,
+				MarkdownDescription: "The monitor timeout in seconds, monitor will fail if it doesn’t complete within this time. Default: `16`",
+			},
+			//"params": jsonObjectSchema("Monitor parameters"),
+			"http": httpMonitorFieldsSchema(),
+			"tcp":  tcpMonitorFieldsSchema(),
+			//"retest_on_failure": schema.BoolAttribute{
+			//	Optional:            true,
+			//	MarkdownDescription: "Enable or disable retesting when a monitor fails. By default, monitors are automatically retested if the monitor goes from \"up\" to \"down\". If the result of the retest is also \"down\", an error will be created, and if configured, an alert sent. Then the monitor will resume running according to the defined schedule. Using retest_on_failure can reduce noise related to transient problems. Default: `true`.",
+			//},
+		},
+	}
+}
+
+//func jsonObjectSchema(doc string) schema.Attribute {
+//	return schema.StringAttribute{
+//		Optional:            true,
+//		MarkdownDescription: fmt.Sprintf("%s. Raw JSON object, use `jsonencode` function to represent JSON", doc),
+//		CustomType:          jsontypes.NormalizedType{},
+//	}
+//}
+
+func statusConfigSchema() schema.Attribute {
+	return schema.SingleNestedAttribute{
+		Optional: true,
+		Attributes: map[string]schema.Attribute{
+			"enabled": schema.BoolAttribute{
+				Optional: true,
+			},
+		},
+	}
+}
+
+func monitorAlertConfigSchema() schema.Attribute {
+	return schema.SingleNestedAttribute{
+		Optional:            true,
+		MarkdownDescription: "Alert configuration. Default: `{ status: { enabled: true }, tls: { enabled: true } }`.",
+		Attributes: map[string]schema.Attribute{
+			"status": statusConfigSchema(),
+			"tls":    statusConfigSchema(),
+		},
+	}
+}
+
+func httpMonitorFieldsSchema() schema.Attribute {
+	return schema.SingleNestedAttribute{
+		Optional:            true,
+		MarkdownDescription: "TODO",
+		Attributes: map[string]schema.Attribute{
+			"url": schema.StringAttribute{
+				Optional:            false,
+				Required:            true,
+				MarkdownDescription: "URL to monitor.",
+			},
+			"ssl_verification_mode": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "Controls the verification of server certificates. ",
+			},
+			"ssl_supported_protocols": schema.ListAttribute{
+				ElementType:         types.StringType,
+				Optional:            true,
+				MarkdownDescription: "List of allowed SSL/TLS versions.",
+			},
+			"max_redirects": schema.StringAttribute{ //TODO: make int64??
```

Comment:
Can this be int64 now? If not can we add the reasoning behind this being a string as a comment?

Reply:
Yep, will do. API returns it as a string, although documents it as a number. I'll do conversion.

Reply:
done.